iointel/src/agent_methods/tools/solscan.py

```
@@ -0,0 +1,741 @@
+import asyncio
+import json
+import os
+from typing import Any, Dict, List, Optional, Literal, Tuple, Annotated
+
+
+import httpx
+
+# Replace these with your actual API key and wallet address
+SOLSCAN_API_KEY = os.getenv("SOLSCAN_API_KEY")
+
+# Base URL for Solscan Pro API V2
+SOLSCAN_API_URL = "https://pro-api.solscan.io"
+
+
+async def fetch_solscan(
+    path, full_url: Optional[str] = None, params: Optional[dict] = None
+) -> dict | str:
+    assert SOLSCAN_API_KEY
```

let's do `raise RuntimeError()` on unset key here, too